### ISSUE TITLE
Change `usize` to `std::num::NonZeroUsize` in `new()` methods

### ISSUE THREAD

**Issue by virtualritz** [greyblake/ta-rs#63]
... and return `Self` instead of `Option<Self>`.

I.e. put the bounds checking that currently happens w/o opt-in inside each indicator's `new()` method at the discretion of the crate user.

**Comment by GeneralGDA** [comment]
Yep, still relevant. Maybe instead of replacing it (as that would be a breaking change), we could create new/alternative "new" versions alongside it?

**Comment by virtualritz** [comment]
TBH I do not think breaking API changes of this nature are an issue in 2026. That's what a Sonnet or even Haiku-level LLM can fix in 10secs.

And if you have tests (which you should 😉 ) they will catch any fallout from this. Although I wouldn't know _what_; since this is a correctness enforcement at the API boundary, not a functional change.

Just my 2c. I do not work in trading any more so I frankly don't care. 😁
